wordpress-mobile/WordPress-Android
# Add Parent Page setting to Page Settings in editor

Fixes #11002 

This PR adds the ability to set the Parent Page via the Page Settings screen inside the editor. 

The approach used here was: 
- Add a new section in the  Page Settings screen for "Page Attributes" and add a setting item inside it called "Parent Page" to keep it consistent with the web editor. 
- Upon clicking this new item, open the already existing `PageParentActivity` for result using the `ActivityLauncher`, which only required a minor refactor in its method signature.
- Upon receiving the result with the new `parentId`, set it to the current post/page, like the other settings.
- Update `PostUtils.postHasEdits` to account for the `parentId` as well.

The `PageParentActivity` is not opened from the `Page Settings` when the device is offline, as that could cause inconsistencies, as pointed out by @ovitrif. Same thing for the `Author` option in the same screen, as it is not able to load the Authors while offline. In this scenario (offline) a snackbar with a "no network" message is shown instead.

For getting the parent title, some changes were also required, since that field is not part of the `PostModel` (see "note" below). The changes needed were:
Fetch the parent page for the post in `EditPostRepository` using `PostStore` whenever the Page Settings screen need it and the title was not retrieved before for that same id. We keep a `ParentPage` instance internal to the Repository acting as a "cache".

> Note: `PostModel` has a field called `parentTitle` but that seems to always return the CURRENT page / post title, not its parent. Upon further investigation this comes directly from the backend response, so I couldn't figure out if this is expected and couldn't rely on this field.

To make sure the "Page Parent" selection screen also loads correctly for newly created pages, a change was done in the `PageParentViewModel` to load the pages for `null` `PageModel`s (`pageId = 0`). This scenario happens when you just create a new page, so you are send to the Editor, and from there you try to set a Page Parent using the new setting. At that point, the page is not even saved locally in the `PageStore` so its ID is 0, but that shouldn't be a problem for setting a parent to the in-memory `PageModel` that is being created. Also, after talking to @ovitrif about the analytics event in the save button for this screen we agreed that it makes sense to skip it for new pages in that state (in-memory).

| Android | Web |
| -------  | ----- |
| ![parent_page_top_level_title](https://user-images.githubusercontent.com/5091503/197286411-717d2828-58a8-4f36-97f9-4ef7dc80d99b.png) | ![image](https://user-images.githubusercontent.com/5091503/195739974-7b9125be-10c0-4185-8436-9bc99e493ad0.png) |

Known pre-existing (likely unexpected) behavior:
While testing my changes I realized that there is a behavior that I'm not sure is documented / expected, which is after making any local changes in a Page (changing anything, including parent in the `Editor` screen, then navigating up to the `Pages` screen without hitting `Update`), if you try to `Set Parent` for that specific page from the `Pages` screen, via the overflow menu in the Page item, it will trigger a full update of that content, not just the `Parent Page`.
This was NOT introduced by this PR, but something I found important to add here, since it's related to Parent Pages.

To test:

0. Open the Wordpress app
1. Select "Pages" in the "Menu" tab
2. Select any existing page
3. Tap the overflow meu (`⋮` - three dots)
4. Select "Page Settings"
   - **Verify** "Parent Page" setting in "Parent Attributes" section
5. Tap the "Parent Page" option
   - **Expect** Parent Page selection screen to show a list of possible parents
6. Tap a new parent from the list
7. Tap the Save button at the top right corner
   - **Verify** you are in the "Page Settings" again and the new page is now selected
8. Hit "Update" to save the changes

After doing that you can verify the change was applied in a few different ways:
- In the "Pages" screen the page you updated will be indented under the new parent page selected
- In the "Pages" screen, follow this steps:
   1. Tap the overflow meu (`⋮` - three dots) for the page you just updated
   2. Tap "Set Parent"
   3. **Verify** the current selection is the new parent
- From the "Pages" screen, follow this steps:
   1. Select the page you just updated
   4. Tap the overflow meu (`⋮` - three dots)
   5. Select "Page Settings"
   6. **Verify** "Parent Page" setting in "Parent Attributes" section has the new parent set

## Regression Notes
1. Potential unintended areas of impact
   - Set Parent Page in the Pages Screen (`PagesFragment`)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
   - Check the `parentId` field in `EditPostRespository`
   - Check the `parentTitle` field population and update in `EditPostRepository`
   - Unit tests for basic existing functionality and new changes in the `PageParentViewModel`

PR submission checklist:
